--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,6 +12,7 @@
 option(ENABLE_TESTS "Enable all tests and checks" OFF)
 option(ENABLE_COVERAGE "Enable coverage reports (includes enabling all tests and checks)" OFF)
 option(ENABLE_WERROR "Treat all build warnings as errors" OFF)
+option(ENABLE_INTROSPECTION "Enable introspection" ON)
 
 if(ENABLE_COVERAGE)
     set(ENABLE_TESTS ON)
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -108,6 +108,8 @@
 
 # AyatanaIdo3-0.4.gir
 
+if (ENABLE_INTROSPECTION)
+
 find_package(GObjectIntrospection REQUIRED QUIET)
 
 if (INTROSPECTION_FOUND)
@@ -183,3 +185,5 @@
     endif ()
 
 endif ()
+
+endif ()
