--- a/configure
+++ b/configure
@@ -14801,6 +14801,8 @@ else
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 	have_valgrind=yes
+fi
+
 fi
  if test "x$have_valgrind" = "xyes"; then
   HAVE_VALGRIND_TRUE=
@@ -14811,8 +14813,6 @@ else
 fi
 
 
-fi
-
 
 
 
--- a/configure.ac
+++ b/configure.ac
@@ -120,8 +120,8 @@ PKG_CHECK_MODULES(DBUSMENUTESTS,  json-glib-1.0 >= $JSON_GLIB_REQUIRED_VERSION
                                   [have_tests=yes]
 )
 PKG_CHECK_MODULES(DBUSMENUTESTSVALGRIND, valgrind, have_valgrind=yes, have_valgrind=no)
-AM_CONDITIONAL([HAVE_VALGRIND], [test "x$have_valgrind" = "xyes"])
 ])
+AM_CONDITIONAL([HAVE_VALGRIND], [test "x$have_valgrind" = "xyes"])
